### DISCUSSION TITLE
Remove index.php from the URL

### DISCUSSION THREAD

**Discussion by smknstd** [laravel/framework#47762] [Q&A]
https://thewebtier.com/remove-index-php-from-the-url-in-laravel

Do you think the laravel framework should handle this ?

ex:
https://spatie.be/index.php/products
https://forge.laravel.com/index.php
https://beyondco.de/index.php/software

**Comment by henzeb** [comment]
Request unclear. You mean: "redirect to the route without the index.php if that is in the url"?

**Reply by smknstd** [reply]
Yes, as explained in the blog post, search engines like google when crawling a Laravel website throws exception (in the search console) and report those urls as duplicated.

**Reply by henzeb** [reply]
Yes, but you should explain it here.... "do you think that laravel framework should handle this" is being lazy.

**Comment by smknstd** [comment]
Here is a given route in a Laravel website:

https://thewebtier.com/articles

However it seems that the following url will also work:

https://thewebtier.com/index.php/articles

This situation can cause that competition could create urls for those unwanted urls with index.php and search engines will see duplicated urls for our whole website and finally we don't know which urls will be displayed in search engines (those without index.php or those with it).

ex:
![Screenshot 2023-07-18 at 11-00-10 spatie be products - Recherche Google](https://github.com/laravel/framework/assets/2412608/8c24b356-1c1e-447e-a735-0595d07ca7e1)

**Comment by smknstd** [comment]
This has been already discussed here, but Apparently there is still no mention at all in the docs or something: https://github.com/laravel/framework/issues/30398
